File: Lv.1/ex35.js
//프로그래머스 제일 작은 수 제거하기
function solution(arr) {
    var answer = [];
    if(arr.length <= 1)
        return [-1];
    else{
        //Math.min()에는 (1,2,3,4) 같이 배열형식이 들어가는게아니다.
        arr.splice(arr.indexOf(Math.min(...arr)),1);
        return arr
    }
    
}
//처음엔 제일 작은수를 정렬로 구했지만, 문제의 조건에서 정렬은 요구하지않기때문에 정렬해버리면 답이틑리게되었다.
//따라서 Math.min()으로 최소값을 구해야하는데 이 과정에서 Math.min()의 인자로 배열을 받지않고 spread연산자 ...로 풀어진
//값들이 들어가야 올바른결과가 나온다는것을 알게되었다.
//그리고 splice함수는 값을 리턴시키는게아니라 원본을 변화시킨다는것을 다시한번 상기시켰다.

File: Lv.1/ex35.java
//K번째수
import java.util.Arrays;

//큰 문제는 없으나 Java 에서 배열을 다루는데에 익숙하지 못하기에 난잡하게 풀어서 해결한것같다, 정수배열의 정렬을 위해
//Arrays.sort()를 떠올려 사용.
public class ex35 {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];
        for(int i=0; i<commands.length; i++){
            //이부분
            int[] arr = new int[commands[i][1]-commands[i][0]+1];
            int idx =0;
            for(int j=commands[i][0]-1; j<commands[i][1]; j++){
                arr[idx++]=array[j];
            }

            //int[] arr = Arrays.copyOfRange(array, commands[i][0]-1, commands[i][1]); 로 대체가 가능하다
            
            Arrays.sort(arr);
            answer[i] = arr[commands[i][2]-1];
        }
        return answer;
    }
}
